ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/interactions.provider.tsx

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) Connexta, LLC */
+import React, { useState } from 'react'
+// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'cesi... Remove this comment to see the full error message
+import Cesium from 'cesium/Build/Cesium/Cesium'
+import Backbone from 'backbone'
+
+export type Translation = {
+  longitude: number
+  latitude: number
+}
+
+export type InteractionsContextType = {
+  interactiveGeo: number | null
+  setInteractiveGeo: (interactiveGeo: number | null) => void
+  interactiveModels: Backbone.Model[]
+  setInteractiveModels: (models: Backbone.Model[]) => void
+  moveFrom: Cesium.Cartographic | null
```

Comment:
I've hardcoded Cesium in here for now, but we should be able to make this context generic so we can reuse it for the 2D map.